src/commands/appSettings/confirmOverwriteSettings.ts

```diff
@@ -0,0 +1,81 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { DialogResponses } from "vscode-azureextensionui";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+
+export async function confirmOverwriteSettings(sourceSettings: { [key: string]: string }, destinationSettings: { [key: string]: string }, destinationName: string): Promise<void> {
+    let suppressPrompt: boolean = false;
+    let overwriteSetting: boolean = false;
+
+    const addedKeys: string[] = [];
+    const updatedKeys: string[] = [];
+    const userIgnoredKeys: string[] = [];
+    const matchingKeys: string[] = [];
+
+    for (const key of Object.keys(sourceSettings)) {
+        if (destinationSettings[key] === undefined) {
+            addedKeys.push(key);
+            destinationSettings[key] = sourceSettings[key];
+        } else if (destinationSettings[key] !== sourceSettings[key]) {
+            if (!suppressPrompt) {
+                const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
+                const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
+                const message: string = localize('overwriteSetting', 'Setting "{0}" already exists in "{1}". Overwrite?', key, destinationName);
+                const result: vscode.MessageItem = await ext.ui.showWarningMessage(message, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
+                if (result === DialogResponses.yes) {
+                    overwriteSetting = true;
+                } else if (result === yesToAll) {
+                    overwriteSetting = true;
+                    suppressPrompt = true;
+                } else if (result === DialogResponses.no) {
+                    overwriteSetting = false;
+                } else if (result === noToAll) {
+                    overwriteSetting = false;
+                    suppressPrompt = true;
+                }
+            }
+
+            if (overwriteSetting) {
+                updatedKeys.push(key);
+                destinationSettings[key] = sourceSettings[key];
+            } else {
+                userIgnoredKeys.push(key);
+            }
+        } else {
+            matchingKeys.push(key);
+        }
+    }
+
+    if (addedKeys.length > 0) {
```

Comment:
If you're going to show all the added keys, might be worth to show the values as well.